runtime/verutil/chverify.c

```diff
@@ -83,8 +83,9 @@ checkNameImpl (J9CfrConstantPoolInfo * info, BOOLEAN isClass, BOOLEAN isMethod,
 			return -1;
 		case ';':
 			if (isClass) {
-				/* Valid at the end of array classes */
-				if (arity && ((c + 1) == end)) {
+				/* Valid at the end of array classes or for valuetypes */
```

Comment:
The comment at the top of the function also needs to be updated

`Classes can end with ';' only if they are array classes.`